cpp/src/DO/Sara/Core/Image/Operations.hpp

```diff
@@ -288,21 +282,17 @@ namespace DO { namespace Sara {
     return dst;
   }
 
-  template <typename T>
-  inline Image<T> safe_crop(const ImageView<T>& src, int top_left_x,
-                            int top_left_y, int width, int height)
-  {
-    auto begin_coords = Vector2i{ top_left_x, top_left_y };
-    auto end_coords = Vector2i{ top_left_x + width, top_left_y + height };
-    return safe_crop(src, begin_coords, end_coords);
-  }
-
-  template <typename T>
-  inline Image<T> safe_crop(const ImageView<T>& src, int center_x, int center_y,
-                            int radius)
+  template <typename T, int N>
+  Image<T, N>
+  safe_crop(const ImageView<T, N>& src,
```

Comment:
Note to self: `safe_crop` implementation is actually wrong and should be done on an infinite image because the SubarrayIterator must be within the image domain bounds. And we can break this pre-condition when we use in `im2col`. So need to reimplement this.